firmware/Makefile

```diff
@@ -2,6 +2,9 @@ PLATFORM =
 PROJECT = 
 
 BUILD_DIR = build/$(PROJECT)/$(PLATFORM)
+CUBEMX_DIR := projects/$(PROJECT)/platforms/$(PLATFORM)/cubemx
```

Comment:
This will fail if `PLATFORM != stm32f767` since other platforms do not have a `cubemx` subdirectory

Reply:
Is the only solution using a condition to check if the project is stm vs something else? Or is there a better approach

Reply:
any code compiled by CMake is placed in the build/ directory, which is why it is so easy to clean most of the project.
We are trying to clean the code generated by CubeMX, and compiled by the generated Makefile. Since CMake doesn't directly compile this code, it goes into the cubemx/build directory

For now, check that the platform starts with stm32. That will be sufficient